docs/NEW_LANGUAGE_FEATURE_INTEGRATION_PROCESS.md

```diff
@@ -0,0 +1,121 @@
+# New Language Feature Check Integration Process [WIP]
+
+## Check Update Procedure
```

Comment:
Add 1-2 sentences here about how updating checks to use modern paradigms and support new language features help us to stay relevant.